R/unnest.R

```diff
@@ -114,7 +114,13 @@ unnest.data.frame <- function(data, ..., .drop = NA, .id = NULL,
     unnested_atomic <- dplyr::bind_cols(unnested_atomic)
   }
 
-  unnested_dataframe <- map(nest_types$dataframe %||% list(), dplyr::bind_rows, .id = .id)
+  unnested_dataframe <- map(nest_types$dataframe %||% list(), ~ {
+    if (length(.) == 0L && !is.null(attr(., "model"))){
+      attr(., "model")
```

Comment:
I think this would be clearer as

```R
if (length(.) == 0L) {
  attr(., "model") %||% data.frame()
} else {
  ...
}
```

And since it's multiple lines, I'd use an anonymous function instead of `~`.